gateway/src/apicast/policy/http_proxy/proxy.lua

```diff
@@ -0,0 +1,48 @@
+local policy = require('apicast.policy')
+local _M = policy.new('http_proxy')
+
+local resty_url = require 'resty.url'
+local ipairs = ipairs
+
+local new = _M.new
+
+local proxies = {"http", "https"}
+
+function _M.new(config)
+  local self = new(config)
+  self.proxies = {}
+
+  if config.all_proxy then
+    local err
+    self.all_proxy, err =  resty_url.parse(config.all_proxy)
+    if err then
+      ngx.log(ngx.WARN, "All proxy '", config.all_proxy, "' is not correctly defined, err:", err)
+    end
+  end
+
+  for _, proto in ipairs(proxies) do
+    local val, err =  resty_url.parse(config[string.format("%s_proxy", proto)])
+    if err then
+      ngx.log(ngx.WARN, proto, " proxy is not correctly defined, err: ", err)
+    end
+    self.proxies[proto] = val or self.all_proxy
+  end
+  return self
+end
+
+local function find_proxy(self, scheme)
+  return self.proxies[scheme]
+end
+
+function _M:export()
+  return  {
+    get_http_proxy = function(_, uri)
```

Comment:
I don't see why this function needs the first param.
Can't we get rid of it and then just call `context.get_http_proxy(self.uri)` from the upstream module instead of `context:get_http_proxy(self.uri)` ?